starter-ui/src/test/java/org/eclipse/starter/ui/StarterServletTest.java

```diff
@@ -0,0 +1,129 @@
+package org.eclipse.starter.ui;
+
+import jakarta.servlet.ServletOutputStream;
+import jakarta.servlet.WriteListener;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Objects;
+import java.util.stream.Stream;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class StarterServletTest {
+
+    private StarterServlet sut;
+
+    @Mock
+    private HttpServletRequest httpServletRequestMock;
+
+    @Mock
+    private HttpServletResponse httpServletResponseMock;
+
+    private ServletOutputStream servletOutputStreamStub;
+
+
+    @BeforeEach
+    void setUp() {
+        this.servletOutputStreamStub = new StubServletOutputStream();
+
+        this.sut = new StarterServlet();
+    }
+
+    @Test
+    void doGet() throws Exception {
+        //given
+        when(this.httpServletRequestMock.getParameter("archetypeGroupId")).thenReturn("org.eclipse.starter");
+        when(this.httpServletRequestMock.getParameter("archetypeArtifactId")).thenReturn("jakartaee10-minimal");
+        when(this.httpServletRequestMock.getParameter("archetypeVersion")).thenReturn("1.1.0");
+        when(this.httpServletRequestMock.getParameter("groupId")).thenReturn("com.sample");
+        when(this.httpServletRequestMock.getParameter("artifactId")).thenReturn("hello_world");
+        when(this.httpServletRequestMock.getParameter("version")).thenReturn("42.0.0-SNAPSHOT");
+        when(this.httpServletResponseMock.getOutputStream()).thenReturn(this.servletOutputStreamStub);
+
+        //when
+        this.sut.doGet(this.httpServletRequestMock, this.httpServletResponseMock);
+
+        //then
+        // Verify mock interactions
+        verify(this.httpServletResponseMock).setContentType("application/zip");
+        verify(this.httpServletResponseMock).setHeader("Content-Disposition", "attachment;filename=\"hello_world.zip\"");
+        verify(this.httpServletResponseMock, times(1)).getOutputStream();
```

Comment:
It's more correct if this is `atLeastOnce()` instead of `times(1)`. But there's no need to verify this as the resulting ZIP is verified later and it doesn't matter how many times `getOutputStream()` was called if the ZIP is OK.

Reply:
true I tend to over test sometimes

Reply:
made it atLeastOnce()